cmd/prepare-storage.go

```diff
@@ -169,28 +168,24 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 		return WaitForFormatting
 	}
 
-	// Total disks unformatted are in quorum verify if we have some offline disks.
-	if disksUnformatted >= quorum {
-		// Some disks offline and some disks unformatted, wait for all of them to come online.
-		if disksUnformatted+disksFormatted+disksOffline == diskCount {
-			return WaitForAll
-		}
-
-		// Some disks possibly corrupted and too many unformatted disks.
-		return Abort
-	}
-
 	// Already formatted and in quorum, proceed to initialization of object layer.
 	if disksFormatted >= readQuorum {
 		if disksFormatted+disksOffline == diskCount {
 			return InitObjectLayer
 		}
 
 		// Some of the formatted disks are possibly corrupted or unformatted, heal them.
-		return WaitForHeal
+		return SuggestToHeal
+	}
+
+	// Some disks unformatted, Some disks formatted and some disks are offline.
+	// This is an undecisive state - wait for all of offline disks to be online
+	// to figure out the course of action.
+	if disksUnformatted+disksFormatted+disksOffline == diskCount {
+		return WaitForAll
 	}
 
-	// Exhausted all our checks, un-handled errors perhaps we Abort.
+	// Exhausted all our checks, un-handled situations such as some disks corrupted we Abort.
```

Comment:
we should stay in `WaitForAll` state even when there are corrupted disks. `disksUnformatted+disksFormatted+disksOffline == diskCount` is not a necessary condition.

E.g, in a 8 disk setup where
```go
disksFormatted = 2
disksOffline = 4
disksCorrupted = 1
disksUnformatted = 1
```
Here we don't have `readQuorum` in formatted disks, so we need to wait. It is still possible that the 2 of the 4 offline disks may be formatted, so we can't abort at this point.